sw/host/rom_ext_image_tools/signer/image/src/image.rs

```diff
@@ -31,51 +38,97 @@ impl RawImage {
+    /// run-time.
+    pub fn update_static_fields(&mut self, config: &ParsedConfig) {
+        // TODO: checks to make sure that the config values (strings) are not
+        //       bigger than the actual field size.
 
-        let mut myclosure = |value, offset| {
+        let mut update = |value, offset| {
             let bytes = str_to_vec_u8(value);
-            let data = &mut self.data;
-            let begin = offset as usize;
-            let end = begin + bytes.len();
-            data.splice(begin..end, bytes.iter().cloned());
+            self.update_field(&bytes, offset);
         };
 
-        myclosure(
+        update(
             &config.manifest_identifier,
             manifest::ROM_EXT_MANIFEST_IDENTIFIER_OFFSET,
         );
-        myclosure(&config.image_length, manifest::ROM_EXT_IMAGE_LENGTH_OFFSET);
-        myclosure(
+        update(
             &config.image_version,
             manifest::ROM_EXT_IMAGE_VERSION_OFFSET,
         );
-        myclosure(
-            &config.image_timestamp,
-            manifest::ROM_EXT_IMAGE_TIMESTAMP_OFFSET,
+
+        update(
+            &config.extensions[0].offset,
+            manifest::ROM_EXT_EXTENSION0_OFFSET_OFFSET,
         );
-        myclosure(
-            &config.extension0_checksum,
+        update(
+            &config.extensions[0].checksum,
             manifest::ROM_EXT_EXTENSION0_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension1_checksum,
+        update(
+            &config.extensions[1].offset,
+            manifest::ROM_EXT_EXTENSION1_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[1].checksum,
             manifest::ROM_EXT_EXTENSION1_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension2_checksum,
+        update(
+            &config.extensions[2].offset,
+            manifest::ROM_EXT_EXTENSION2_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[2].checksum,
             manifest::ROM_EXT_EXTENSION2_CHECKSUM_OFFSET,
         );
-        myclosure(
-            &config.extension3_checksum,
+        update(
+            &config.extensions[3].offset,
+            manifest::ROM_EXT_EXTENSION3_OFFSET_OFFSET,
+        );
+        update(
+            &config.extensions[3].checksum,
             manifest::ROM_EXT_EXTENSION3_CHECKSUM_OFFSET,
         );
+
+        self.update_usage_constraints_field(&config.input_files.usage_constraints_path);
+
+        self.update_peripheral_lockdown_info_field(&config.peripheral_lockdown_info);
+
+        // TODO: calculated at runtime, so we should probably rename this
+        //       function.
+        self.update_timestamp_field();
+    }
+
+    /// Updates ROM_EXT manifest signature key public exponent field.
+    pub fn update_exponent_field(&mut self, exponent: &[u8]) {
+        let offset = manifest::ROM_EXT_SIGNATURE_KEY_PUBLIC_EXPONENT_OFFSET;
```

Comment:
Please inline this. This variable doesn't make the code more readable.

Reply:
It might be in a style guide somewhere, but given that >80c just means work for rustfmt and not you I'm in agreement with @mcy here.

Reply:
Done.